entoas/spec/marshal.go

```diff
@@ -0,0 +1,143 @@
+package spec
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+)
+
+func (spec Spec) MarshalJSON() ([]byte, error) {
+	type Local Spec
+	return json.Marshal(struct {
+		Local
+		Version string `json:"openapi"`
+	}{
+		Local:   Local(spec),
+		Version: version,
+	})
+}
+
+func (p ParameterPlace) MarshalJSON() ([]byte, error) {
+	switch p {
+	case InQuery:
+		return json.Marshal("query")
+	case InHeader:
+		return json.Marshal("header")
+	case InPath:
+		return json.Marshal("path")
+	case InCookie:
+		return json.Marshal("cookie")
+	default:
+		return nil, fmt.Errorf("cannot marshal %v to json", p)
+	}
+}
+
+func (p *ParameterPlace) UnmarshalJSON(d []byte) error {
+	var s string
+	if err := json.Unmarshal(d, &s); err != nil {
+		return err
+	}
+	switch s {
+	case "InQuery":
+		*p = InQuery
+	case "InHeader":
+		*p = InHeader
+	case "InPath":
+		*p = InPath
+	case "InCookie":
+		*p = InCookie
+	default:
+		return fmt.Errorf("cannot unmarshal %v to ParameterPlace", p)
+	}
+	return nil
+}
+
+func (f Field) MarshalJSON() ([]byte, error) {
+	type local Field
+	j, err := json.Marshal(local(f))
+	if err != nil {
+		return nil, err
+	}
+	if f.Unique {
+		return j, nil
+	}
+	return []byte(fmt.Sprintf(`{"type":"array","items":%s}`, string(j))), nil
```

Comment:
```suggestion
	return []byte(fmt.Sprintf(`{"type":"array","items":%s}`, j)), nil
```

conversion is not needed.